src/qcodes/parameters/delegate_parameter.py

```diff
@@ -314,3 +350,18 @@ def validate(self, value: ParamDataType) -> None:
         super().validate(value)
         if self.source is not None:
             self.source.validate(self._from_value_to_raw_value(value))
+
+    @property
+    def validators(self) -> tuple[Validator, ...]:
+        """
+        Tuple of all validators associated with the parameter. Note that this
+        includes validators of the source parameter if source parameter is set
+        and has any validators.
+
+        :getter: All validators associated with the parameter.
+        """
+        source_validators: tuple[Validator, ...] = (
+            self.source.validators if self.source is not None else ()
+        )
+
+        return tuple(self._vals) + source_validators
```

Comment:
just wonder, if it's cleaner (and even possible) to use super here?
```suggestion
        return super().validators + source_validators
```